### DISCUSSION TITLE
Legal question for commercial use

### DISCUSSION THREAD

**Discussion by whyvas** [mapsforge/mapsforge#1262] [Q&A] [answered]
Good day devemux86, 

First of all, thank you so much for all your work on this project, it is really amazing and extremely useful software. 

I've written an Android application that uses the mapsforge library and I am trying to get my employer to commercialize the software as a proprietary, closed-source application. Our legal department is not so keen on the idea and has provided the following concern: 

_As indicated on the GitHub site, Mapsforge is licensed under the LPGL v3 license, except that subsections 4(d) and 4(e) are waived. As a result, the requirement to dynamically link the Mapsforge library is removed, and the requirement to release all the source code is removed. However, you still do have to find a way to comply with the first sentence of section 4 of the LPGL v3 license (You may convey a Combined Work under terms of your choice that, taken together, effectively do not restrict modification of the portions of the Library contained in the Combined Work and reverse engineering for debugging such modifications), with the ultimate goal of enabling modification of the Mapsforge code in the context of your application.

 In other words, the waiver of section 4(d) has not removed the requirement to provide the Mapsforge code so that it can be modified then used – it has simply removed any reference to how to provide the Mapsforge code._

Based on the large number of applications listed as Proprietary/Commercial and closed source on your "Applications using mapsforge software" page, I suspect the above is not the correct interpretation of the waiver but I figure it would be best to simply ask the maintainer directly. 

In order to comply with your license terms:
Do we have to include a way for users to be able to modify the mapsforge library used within our application? 
Does the library have to be dynamically linked so that it can be modified independently of the main application?

Thanks again for the amazing product!

**Comment by devemux86** [comment, marked as answer]
[LGPL v3 license](https://www.gnu.org/licenses/lgpl-3.0) states:

_"You may convey a Combined Work under terms of your choice that, taken together, effectively do not restrict modification of the portions of the Library contained in the Combined Work and reverse engineering for debugging such modifications, **if you also do each of the following:**"_

The expression in bold depends on subsequent sections, but the mentioned ones are waived to allow use on Android.

---

Finally the library mentions:

_"This means that you are allowed to convey a Combined Work without providing the user any way to recombine or relink the application and without providing any shared library mechanism.
In other words, **you are allowed to include Mapsforge library in your Android application**, without making your application open source."_

**Comment by whyvas** [comment]
Thank you for the prompt response! Acknowledging that we can keep it closed source, the legal department still believes that we have to dynamically link the library or provide another method so that users can modify mapsforge for use within my application (I don't think this is even possible with an Android APK), can you confirm that there are no legal requirements to provide users of the software with the ability to modify the mapsforge library in any way?

I am terribly sorry to bother you with this request.

**Reply by devemux86** [reply]
You can also check discussions with original authors in the old forum:
https://groups.google.com/g/mapsforge-dev/c/WWJVjZwc2OM/m/F9We-R5QgEwJ
